Client.py

```diff
@@ -0,0 +1,227 @@
+                self._send_buff = All the data that needed to be sent but
+                                  hadn't sent yet
+                self._buff_size = The buff size of the server
+                self.file = FileObject of the file the request wants
+                self.request = Request object of the request we got
+    """
+
+    def __init__(
+        self,
+        s,
+        buff_size,
+        file_object=None,
+        request=None
+    ):
+        """
+        Client of Server, handle everything by itself
+        Arguements:
+        s: the socket of this client, CANNOT BE MODIFIED
+        _recv_buff: Every data we recv from the client and had not been
+                    procssed (mostly partial lines)
+        _send_buff: Every data we need to send but didn't sent yet
+        """
+        super(Client, self).__init__()
+        self._socket = s
+        self._recv_buff = ""
+        self._send_buff = ""
+        self._buff_size = buff_size
+        self.file = file_object
+        self.request = request
+
+    def get_socket(self):
+        """Get the socket of this client, private arguement"""
+        return self._socket
+
+    def _parse_header(self, line):
+        parsed = line.split(":", 1)
+        if len(parsed) == 2:
+            self.request.add_header(*parsed)
+            self.logger.debug("Added header, %s", line)
+
+    def recv(self):
+        """Recv data from the client socket and process it to Reqeust and
+           FileObject, or put it in the buff"""
+        try:
+            line = ""
+            # If there is no request, get request line
+            if self.request is None and constants.CRLF not in self._recv_buff:
+                self._recv_lines()
+                self.logger.debug("after recv %s", self._recv_buff)
+            # If we have the request line, but don't have request object, creat
+            # one
+            if self.request is None and constants.CRLF in self._recv_buff:
+                parsed_lines = self._recv_buff.split(constants.CRLF, 1)
+                req = parsed_lines[0].split(' ', 2)
+                self.logger.debug("Req, %s %s %s", req)
+                if req[2] != constants.HTTP_VERSION:
+                    raise RuntimeError('Not HTTP protocol')
+                if len(req) != 3:
+                    raise RuntimeError('Incomplete HTTP protocol')
+                method, uri, signature = req
+                if method not in SUPPORTED_METHODS:
+                    raise RuntimeError(
+                        "HTTP unspported method '%s'" % method)
+                if not uri or uri[0] != '/' or '\\' in uri:
+                    raise RuntimeError("Invalid URI")
+                file_name = os.path.normpath(
+                    '%s%s' % (constants.BASE, os.path.normpath(uri)))
+                self.file = FileObject.FileObject(file_name)
+                self.request = Request.Request(method, uri)
```

Comment:
It can be but it better as objects.
Object will allow me to work better with the headers in the future because I'll be able to write functions only for the request. 
Also it allow me to keep thing in private, while I can't do it in dictionary.